tasks/friends-app.md

```diff
@@ -0,0 +1,61 @@
+[![MIT Licensed][icon-mit]][license]
+[![Awesome][icon-awesome]][awesome]
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
+[![Telegram][icon-chat]][chat]
+
+# Friends App
+
+When you're done with your challenge push your code to github repo and publish
+it using `gh-pages`. _Add link to it to your main frontend tasks repo._
+
+If you honestly finished all the previous steps than go ahead and share it with
+others - send a message in [course channel](chat) with the link to you repo and the words "Weather APP - finished" and add link to repo with code of the
+website.
+
+## Friends app
+
+Create a tiny social friends' search page with users cards, search, sorting and filtering of them by age, name or whatewer you want with [Random User API](https://randomuser.me) :)
+![image](https://i.imgur.com/5tcpqcY.png).
+
+### As a user, I would like to:
+
+* See list of my mock friends :)
+* can search them by typing name in search input and see immediately filtering on the page
+* sort cards by name/age in A-Z/Z-A ways
+* filter friends by age, by name, whatever you want else options
+
+- [ ] you can use all `Array.methods` which you know :) for instance - Array.sort (text sort, number sort) `/filter/slice/splice/shift/map/reduce`
+- [ ] you'll try to figure out with request/response data incapsulation and try callbacks' logic in action
+- [ ] understand sync/async `GET request`, params & queries in action, something one of `ajax/fetch/xhr/`etc
+- [ ] understand pros & cons of the immutable fundamentals for complex types aka array on example of copies of arrays and how manage them
+- [ ] you can try form' key/mouse events, manage them, enable and disable them
+
+### Stuff which will help you:
+
+* [Основы XMLHttpRequest](https://learn.javascript.ru/ajax-xmlhttprequest) - you
+  can use XMLHttpRequest for sending requests to webserver;
+* [Randomuser api with awesome simple examples and docs](https://randomuser.me) -
+  _you can use one of this API for getting data for your website_;
+* [URL API](https://developer.mozilla.org/en-US/docs/Web/API/URL) - you can use
+  this for working with URL;
+* Take a look on [Array methods on learn.javascript.ru](https://learn.javascript.ru/array-methods)
+* make good responsive/adaptive/fluid/elastic/whatever design and other modern stuff aka [Materialize](https://materializecss.com/), [MD lite](https://getmdl.io/), [Bootstrap 3/4](https://getbootstrap.com/) etc...
```

Comment:
As for me, they should use materialize, etc. as later as possible.

Reply:
I can delete these CSS frameworks but it makes the task harder with custom styles. As you wish

Reply:
Yes, delete pls. It is not so hard to create some basic layout.

Reply:
done